packages/elements/src/photon-medication-search/photon-medication-search-component.tsx

```diff
@@ -0,0 +1,350 @@
+// Solid
+import { customElement } from 'solid-element';
+import { createMemo, createSignal, onMount, Show } from 'solid-js';
+
+// Photon
+import { usePhoton } from '../context';
+import { PhotonMedicationDropdownFullWidth } from '../photon-medication-dropdown-full-width';
+import { PhotonMedicationDropdown } from '../photon-medication-dropdown';
+import { PhotonDropdown } from '../photon-dropdown';
+
+// Types
+import {
+  Medication,
+  Treatment,
+  PrescriptionTemplate,
+  TreatmentOption
+} from '@photonhealth/sdk/dist/types';
+import { CatalogStore } from '../stores/catalog';
+
+import { ApolloClient } from '@apollo/client';
+import gql from 'graphql-tag';
+
+// Utility Functions
+
+function triggerCustomEvent(ref: any, eventName: string, detail: any) {
+  const event = new CustomEvent(eventName, {
+    composed: true,
+    bubbles: true,
+    detail
+  });
+  ref?.dispatchEvent(event);
+}
+
+function dispatchTreatmentSelected(
+  ref: any,
+  datum: Treatment | PrescriptionTemplate,
+  catalogId: string
+) {
+  triggerCustomEvent(ref, 'photon-treatment-selected', { data: datum, catalogId });
+}
+
+type DataReturn<Type> = {
+  data?: Type;
+  loading?: boolean;
+  error?: any;
+  errors?: readonly any[];
+};
+
+const SearchTreatmentoptions = gql`
+  query SearchTreatmentOptions($searchTerm: String!) {
+    treatmentOptions(searchTerm: $searchTerm) {
+      medicationId
```

Comment:
To make things easier, I think it'd probably do `id: medicationId` so that you can use `id` in all the various places that we already have wired up for `id`

Reply:
done